Firestore/core/src/firebase/firestore/util/CMakeLists.txt

```diff
@@ -16,69 +16,66 @@
 # libraries in here are an implementation detail of making this a
 # mutli-platform build.
 
-add_library(
+cc_library(
   firebase_firestore_util_base
-  secure_random_arc4random.cc
-  string_printf.cc
-)
-target_link_libraries(
-  firebase_firestore_util_base
-  PUBLIC
-  absl_base
+  SOURCES
+    secure_random.h
+    secure_random_arc4random.cc
+    string_printf.cc
+    string_printf.h
+  DEPENDS
+    absl_base
 )
 
+## assert and log
+
 # stdio-dependent bits can be built and tested everywhere
-add_library(
-  firebase_firestore_util_stdio
-  assert_stdio.cc
-  log_stdio.cc
-)
-target_link_libraries(
+cc_library(
   firebase_firestore_util_stdio
-  PUBLIC
-  firebase_firestore_util_base
+  SOURCES
+    assert_stdio.cc
+    log_stdio.cc
+  DEPENDS
+    firebase_firestore_util_base
+    absl_base
 )
 
 # apple-dependent bits can only built and tested on apple plaforms
-if(APPLE)
-  add_library(
-    firebase_firestore_util_apple
+cc_library(
+  firebase_firestore_util_apple
+  SOURCES
     assert_apple.mm
     log_apple.mm
-  )
-  target_compile_options(
-    firebase_firestore_util_apple
-    PRIVATE
-    ${OBJC_FLAGS}
-  )
-  target_link_libraries(
-    firebase_firestore_util_apple
-    PUBLIC
+    string_apple.h
+  DEPENDS
     FirebaseCore
-  )
-endif(APPLE)
-
-add_library(
-  firebase_firestore_util
-  autoid.cc
+)
+target_compile_options(
```

Comment:
Update: Ah, I see you've already handled this in the next commit. I'll leave my comment here, as it's an alternative to consider, but I think yours is just fine too. No action required.

----

Since this really only applies to the above cc_library, can we move it right into it like you've done with DEPENDS? i.e. something like this:

```
cc_library(
  firebase_firestore_util_apple
  SOURCES
    ...
  DEPENDS
    ...
  COMPILE_OPTIONS
    PRIVATE
    ${OBJC_FLAGS}
)
```

Or something like that. (No idea if that's possible with cmake or not.)

If you do elect to do this (assuming it's even possible) then I don't think it needs to be part of this PR (though I have no objections to that either.)

Reply:
I'd prefer to avoid having to think about this and get this right so I'll stick with the automatic approach.